lockfile_generator/src/lib.rs

```diff
@@ -50,8 +51,15 @@ pub trait Generator {
         command.stderr(Stdio::null());
 
         // Provide better error message, including the failed program's name.
-        let mut child =
-            command.spawn().map_err(|err| Error::ProcessCreation(format!("{command:?}"), err))?;
+        let mut child = command.spawn().map_err(|err| {
+            // Collect program and args for error message.
+            let mut cmdline = format!("{:?}", command.get_program());
+            for arg in command.get_args() {
+                let _ = write!(cmdline, " {arg:?}");
+            }
+
+            Error::ProcessCreation(cmdline, err)
```

Comment:
After some more thought and a conversation with @furi0us333, I actually think that including the args is unnecessary (and potentially misleading)

```suggestion
            Error::ProcessCreation(format!("{:?}", command.get_program()), err)
```

For example, looking at this error (from this branch):

```
❗ Error: failed to run command "pip-compile" "-o" "requirements-locked.txt": No such file or directory (os error 2)
```

Without context, a user may think "No such file" refers to `requirements-locked.txt` (which indeed does not exist). It's cleaner without the args:

```
❗ Error: failed to run command "pip-compile": No such file or directory (os error 2)
```

And as far as I know, the args will never cause `spawn()` to fail. So I think it's safe to ignore them here.